pkg/inmemorychannel/dispatcher.go

```diff
@@ -52,39 +52,45 @@ func (d *InMemoryDispatcher) UpdateConfig(config *multichannelfanout.Config) err
 
 // Start starts the inmemory dispatcher's message processing.
 // This is a blocking call.
-func (d *InMemoryDispatcher) Start(stopCh <-chan struct{}) error {
-	d.logger.Info("in memory dispatcher listening", zap.String("address", d.server.Addr))
+func (d *InMemoryDispatcher) Start(ctx context.Context) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	errCh := make(chan error, 1)
 	go func() {
-		err := d.server.ListenAndServe()
-		if err != nil {
-			d.logger.Error("Failed to ListenAndServe.", zap.Error(err))
-		}
+		errCh <- d.ceClient.StartReceiver(ctx, d.handler.ServeHTTP)
 	}()
-	<-stopCh
-	ctx, cancel := context.WithTimeout(context.Background(), d.server.WriteTimeout)
-	defer cancel()
 
-	err := d.server.Shutdown(ctx)
-	if err != nil {
-		d.logger.Error("Shutdown returned an error", zap.Error(err))
+	// Stop either if the receiver stops (sending to errCh) or if the context Done channel is closed.
+	select {
+	case err := <-errCh:
+		return err
+	case <-ctx.Done():
+		break
+	}
+
+	// Done channel has been closed, we need to gracefully shutdown h.ceClient. cancel() will start its
```

Comment:
```suggestion
	// Done channel has been closed, we need to gracefully shutdown h.ceClient.cancel() will start its
```

Reply:
updated the comment as it was misleading.